_playwright-tests/UI/RepositoryPackages.spec.ts

```diff
@@ -0,0 +1,95 @@
+    await navigateToRepositories(page);
+    await closePopupsIfExist(page);
+    await deleteAllRepos(page);
+
+    await test.step('Create a repository', async () => {
+      await page.getByRole('button', { name: 'Add repositories' }).first().click();
+      await expect(page.getByRole('dialog', { name: 'Add custom repositories' })).toBeVisible();
+      await page.getByLabel('Name').fill(`${repoName}`);
+      await page.getByLabel('Snapshotting').click();
+      await page.getByLabel('URL').fill('https://jlsherrill.fedorapeople.org/fake-repos/signed/');
+      await page.getByRole('button', { name: 'Save', exact: true }).click();
+    });
+
+    await test.step('Wait for status to be "Valid"', async () => {
+      const row = await getRowByNameOrUrl(page, repoName);
+      await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+    });
+
+    await test.step('Verify the package count matches the snapshot', async () => {
+      const row = await getRowByNameOrUrl(page, repoName);
+      const packageCountValue = await row.getByTestId('package_count_button').textContent();
+      await page.getByRole('button', { name: 'Kebab toggle' }).click();
+      await page.getByRole('menuitem', { name: 'View all snapshots' }).click();
+      // click on the first row to view the snapshot details
+      const snapshotPackagesColumn = await page
+        .getByTestId('snapshot_package_count_button')
+        .textContent();
+      // assert the package count matches the after snapshot packages count
+      expect(packageCountValue).toBe(snapshotPackagesColumn);
+      await page.getByText('Close').click();
+    });
+
+    // Edit the repository to change number of packages
+    await test.step('Update the repository', async () => {
+      const row = await getRowByNameOrUrl(page, repoName);
+      await row.getByLabel('Kebab toggle').click();
+      await row.getByRole('menuitem', { name: 'Edit' }).click();
+      await page.getByPlaceholder('Enter name', { exact: true }).fill(edited_repo);
+      await page
+        .getByLabel('URL')
+        .fill('http://jlsherrill.fedorapeople.org/fake-repos/needed-errata/');
+      await page.getByRole('button', { name: 'Save changes', exact: true }).click();
+      const edited_row = await getRowByNameOrUrl(page, edited_repo);
+      await expect(edited_row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+    });
+
+    await test.step('Verify the package count matches the edited snapshot', async () => {
+      const edited_row = await getRowByNameOrUrl(page, edited_repo);
+      const edited_repo_packageCountValue = await edited_row
+        .getByTestId('package_count_button')
+        .textContent();
+      await page.getByRole('button', { name: 'Kebab toggle' }).click();
+      await page.getByRole('menuitem', { name: 'View all snapshots' }).click();
+      // refresh page to get the updated snapshot
+      await page.reload();
```

Comment:
are you really not getting the latest snapshot in that modal already? you shouldn't need to reload the page I think 🤔

Reply:
Until yesterday, snapshots appeared immediately after the repo was edited. 
Bur I’ve noticed that now there's a delay before the changes are reflected. 
Is that might be a bug ?

Reply:
yeah, that's might be a bug, we might be forgetting to invalidate some of the queries 🤔

Reply:
made a [PR](https://github.com/content-services/content-sources-frontend/pull/526) that should fix this, we weren't invalidating some things properly 😅

Reply:
Tried out this the fix but the page still needs to get reloaded

Reply:
hrm 🤔 that's weird, I tested it with this PR and manually when making those changes and it seemed to work 💭 
could you try again and let me know 😇  
if it won't work we should merge this and I will make a ticket to have another look into the cause of this 😅

Reply:
yeah i varified it again and it is not fixing the issue .

[Screencast from 2025-05-13 15-57-55.webm](https://github.com/user-attachments/assets/791a58b7-f983-4cc5-92b2-3adbb2195bcb)

Reply:
Its working now !
i dont have to relaod the page and it started working after merging the fix !!

Reply:
awesome! 🥳